### DISCUSSION TITLE
What is a good protocol to decide hyper-parameters for a domain generalization algorithm?

### DISCUSSION THREAD

**Discussion by smilesun** [marrlab/DomainLab#75] [Ideas]
Different from domain adaptation,  domain generalization requires a robust model against distribution shift before any target domain distribution matching can be done. 

Usually, hyper-parameter tunining is done on a small validation set, since we have no access to target domain data before model has to be deployed, the validation set we can make can only come from the source domains. So does this kind of validation make sense? 

One argument is we also want an algorithm which performs well on the source domain, and a train-validation split on the source domain would be a good way to find a good model that works on the source domain well, but is this better than if we just use training data loss to choose hyper-parameters?
